airflow/models/baseoperator.py

```diff
@@ -16,6 +16,8 @@
 # specific language governing permissions and limitations
 # under the License.
 """Base operator for all operators."""
+from __future__ import annotations
```

Comment:
We don't use yet __future__ annotations. We discussed that we want to do it, but in order to make cherry-picking easier, we will globally add them just before we release 2.4 to not make cherry-picking more difficult than it should be.

https://lists.apache.org/thread/81fr042s5d3v17v83bpo24tnrr2pp0fp

Can you please remove those for now?